types/types.go

```diff
@@ -200,6 +200,23 @@ type ImageInspectInfo struct {
 	Layers        []string
 }
 
+// DockerAuthConfig contains authorization information for connecting to a registry.
+// FIXME: forked from docker/docker, waiting a full docker re-vendor
+type DockerAuthConfig struct {
+	Username string
+	Password string
+	Auth     string
+	// FIXME: currently not used
+	ServerAddress string
+	// IdentityToken is used to authenticate the user and get
+	// an access token for the registry.
+	// FIXME: currently not used
+	IdentityToken string
+	// RegistryToken is a bearer token to be sent to a registry
+	// FIXME: currently not used
+	RegistryToken string
```

Comment:
So, grepping `docker/docker`:
- `RegistryToken` is set nowhere in the code (but could be provided by a third-party client using the Remote API)
- `IdentityToken` is `refresh_token` from https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md , something we neither use nor generate (but we could, I suppose)
- `ServerAddress` seems to be a data structure artifact (look how `cliconfig/configfile/file.go:LoadFromReader` overrides the value in the unmarshaled struct) and/or a semantic mix-up (`registry/service.go` uses `AuthConfig.ServerAddress` to look up registry hostnames)
- `Auth` also an artifact of reusing the same structure for internal data and config storage, it is not really used anywhere, except `cliconfig/configfile/file.go`, which decodes the field _and zeroes it_.

Grepping `openshift/origin`:
- No use of `ReghistryToken`, `IdentityToken`
- AFAICS from a _very_ imprecise grep the only real use is in `vendor/k8s.io/kubernetes/pkg/credentialprovider/keyring.go` , which only sets `Username`, `Password`, `Email`; I might have missed something.